app/static/js/initialize_map.js

```diff
@@ -0,0 +1,119 @@
+geocoder = new google.maps.Geocoder();
+address = document.getElementById("address");
+
+// Global array for map markers
+var markers = []
+
+// Global Marker Wrappers and Map
+var globalMarkerWrappers = null;
+var globalMap = null;
+
+// Initial map center coordinates
+INITIAL_CENTER_LAT = 39.952;
+INITIAL_CENTER_LONG = -75.195;
+initial_coords = new google.maps.LatLng(INITIAL_CENTER_LAT, INITIAL_CENTER_LONG);
+
+// Initial map zoom value
+ZOOM = 17;
+
+// Bounds for the Date Slider. End date is current date.
+var BOUNDS_MIN = new Date(2015, 0, 1);
+var BOUNDS_MAX = new Date();
+
+function MarkerWrapper(actualMarker, incidentDate, contentString) {
+    this.actualMarker = actualMarker;
+    this.incidentDate = incidentDate;
+    this.contentString = contentString;
+}
+
+function storeMarkerState(markerWrappers, map, minDate) {
+    globalMarkerWrappers = markerWrappers;
+    globalMap = map;
+    console.log(markerWrappers[0].actualMarker);
+    for (mw = 0; mw < globalMarkerWrappers.length; mw++)
+    {
+        (globalMarkerWrappers[mw].actualMarker).setMap(globalMap);
+    }
+    BOUNDS_MIN = minDate;
+}
+
+// Use Google geocoder to update geolocation given an address through
+// the address search box
+function update_center() {
+    geocoder = new google.maps.Geocoder();
+    address = $("#address").val();
+    if (geocoder) {
+        console.log(address);
+        geocoder.geocode( { 'address': address},
+        function(results, status) {
+            if (status != google.maps.GeocoderStatus.OK) {
+                alert("Geocode not successful - " + status);
+            }
+            else if (status == google.maps.GeocoderStatus.ZERO_RESULTS) {
+                alert("Address not found!");
```

Comment:
Can we do this in a more semantic-ui-ish type of way

http://semantic-ui.com/collections/message.html#negative--error